internal/interfaces/grpc/handler/wallet.go

```diff
@@ -162,3 +162,17 @@ func (w *wallet) GetInfo(ctx context.Context, _ *pb.GetInfoRequest) (*pb.GetInfo
 		},
 	}, nil
 }
+
+func (w *wallet) Auth(
+	ctx context.Context,
+	req *pb.AuthRequest,
+) (*pb.AuthResponse, error) {
+	verified, err := w.appSvc.Auth(ctx, req.GetPassword())
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Let's add a sanity check for the given pwd:
```suggestion
        password, err := parsePassword(req.GetPassword())
        if err != nil {
                 return nil, status.Error(codes.InvalidArgument, err.Error())
        }
        
	verified, err := w.appSvc.Auth(ctx, req.GetPassword())
	if err != nil {
		return nil, err
	}
```

Reply:
Done